src/console/widget.ts

```diff
@@ -134,6 +134,37 @@ class ConsoleWidget extends Widget {
     // Create the prompt.
     this.newPrompt();
 
+    // Handle inputs/outputs initated by another session.
+    this._session.kernel.iopubMessage.connect((kernel, val) => {
+      let value = val as any;
+      if (value.msg_type != 'execute_input') {
+        return;
+      }
+      console.log(value);
+      // Check whether this message came from an external session.
+      console.log(value.header.session);
+      if (value.parent_header.session === this.session.kernel.clientId) {
+        console.log('message from this session');
+        return;
+      }
+      console.log('message from exernal session');
+      let old_prompt = this.prompt;
+      this.newPrompt();
+      let new_prompt = this.prompt;
+      // Copy any not-yet-executed content into a new prompt.
+      new_prompt.model.source = old_prompt.model.source;
+      // And fill the old prompt with the content from the remote session.
+      console.log('old_prompt.model before');
+      console.log(old_prompt.model);
+      old_prompt.model.source = value.content.code;
```

Comment:
I believe the signals for the old prompt are disconnected when you create a new one.
I am burned out from bundling and CSS this week, but can take a deeper look at some other point ;).